lib/internal/test_runner/test.js

```diff
@@ -29,12 +31,27 @@ const kMultipleCallbackInvocations = 'multipleCallbackInvocations';
 const kParentAlreadyFinished = 'parentAlreadyFinished';
 const kSubtestsFailed = 'subtestsFailed';
 const kTestCodeFailure = 'testCodeFailure';
+const kTestTimeoutFailure = 'testTimeoutFailure';
 const kDefaultIndent = '    ';
 const noop = FunctionPrototype;
 const isTestRunner = getOptionValue('--test');
 const testOnlyFlag = !isTestRunner && getOptionValue('--test-only');
 // TODO(cjihrig): Use uv_available_parallelism() once it lands.
 const rootConcurrency = isTestRunner ? cpus().length : 1;
+const rootTimeout = 30_0000;
+
+
+function PromiseTimeout(promise, timeout) {
```

Comment:
Can you name this something else please - `PromiseTimeout` makes it seem like a primordial. The function name should also be camelCase. `testTimeout()` is the first thing that comes to mind, but you don't have to use that if you don't like it.